pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -129,6 +130,19 @@ func (rc *reconciler) syncStates() {
 	}
 }
 
+func (rc *reconciler) isExclusiveVolume(volumeSpec *volume.Spec) bool {
+	if volumeSpec.PersistentVolume == nil {
+		return false
+	}
+	// check if this volume can be attached to multiple PODs/nodes, if yes, return false
+	for _, accessMode := range volumeSpec.PersistentVolume.Spec.AccessModes {
```

Comment:
GCE pd is allowed to attach to multiple nodes if readOnly is true. See the link for "Share a persistent disk between multiple instances". 
https://cloud.google.com/compute/docs/disks/add-persistent-disk#create_disk

PV's AccessMode cannot be used reliably to determine attachment mode. It is only used for matching PV and PVC.  @saad-ali and I also discussed a little about this. We think it might not always valid even we use ReadOnly to determine whether node could be attached to multiple nodes. Currently for GCE and AWS this is true. But we have other volume types such as Cinder, Vsphere. We have to check those types and also in the future this might change. 
We might need to think some other way to solve the problem.